db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/SelectNode.java

```diff
@@ -2626,4 +2632,21 @@ public Visitable projectionListPruning(boolean considerAllRCs) throws StandardEx
 
         return this;
     }
+
+    /**
+     * Is the GROUP BY clause nonexistent or a dummy constant value?
+     */
+    public boolean hasNoGroupBy(){
+        if (groupByList == null)
```

Comment:
It is probably better to use "if (gorupByList == null || groupByList.isEmpty())".